go/tools/builders/env.go

```diff
@@ -162,6 +163,86 @@ func (e *env) runCommandToFile(out, err io.Writer, args []string) error {
 	return runAndLogCommand(cmd, e.verbose)
 }
 
+func symlinkBuilderCC() (func(), string, error) {
+	absPath, err := filepath.Abs(os.Args[0])
+	if err != nil {
+		return nil, "", err
+	}
+
+	dirname, err := os.MkdirTemp("", "")
+	if err != nil {
+		return func() {}, "", err
+	}
+
+	// re-use the file extension if there is one (mostly for .exe)
+	builderPath := filepath.Join(dirname, "builder-cc"+filepath.Ext(os.Args[0]))
+
+	if err := os.Symlink(absPath, builderPath); err != nil {
+		_ = os.RemoveAll(dirname)
+		return func() {}, "", err
```

Comment:
This cleanup func should remove the directory.

Reply:
what I was going for here was that the cleanup function is only needed if the function succeeds. the code is a little inconsistent in that it returns nil above for the cleanup function and non-nil here, but also returns a non-nil err.

what probably makes the most sense is just to make it consistent and return a nil cleanup function if err is non-nil. the directory should always be removed in case of errors otherwise the caller would have to *also* call the cleanup function on an error which doesn't feel right.

i'll make that change, let me know what you think